File: app/src/main/java/com/kakacat/minitool/todayinhistory/Presenter.java
package com.kakacat.minitool.todayinhistory;

import com.kakacat.minitool.common.myinterface.HttpCallback;
import com.kakacat.minitool.common.util.ThreadUtil;
import com.kakacat.minitool.todayinhistory.model.Article;
import com.kakacat.minitool.todayinhistory.model.Model;

import java.util.List;

import okhttp3.Response;

public class Presenter implements Contract.Presenter {

    private Contract.View view;
    private Model model;

    public Presenter(Contract.View view) {
        this.view = view;
        this.model = Model.getInstance();
    }

    @Override
    public void initData() {
        model.initData();
    }

    @Override
    public void refreshData() {
        model.sendRequest(new HttpCallback() {
            String result = "请求错误";
            boolean needRefresh = false;

            @Override
            public void onSuccess(Response response) {
                if (!model.handleHistoryResponse(response)) {
                    result = "处理失败";
                } else {
                    result = "处理成功";
                    needRefresh = true;
                }
                ThreadUtil.callInUiThread(() -> view.onUpdateDataCallBack(result, needRefresh));
            }

            @Override
            public void onError() {
                ThreadUtil.callInUiThread(() -> view.onUpdateDataCallBack(result, needRefresh));
            }
        });
    }


    @Override
    public List<Article> getArticleList() {
        return model.getArticleList();
    }

    public int getYear() {
        return model.getYear();
    }

    public int getMonth() {
        return model.getMonth();
    }

    public void setMonth(int month) {
        model.setMonth(month);
    }

    public int getDay() {
        return model.getDay();
    }

    public void setDay(int day) {
        model.setDay(day);
    }
}

File: app/src/main/java/com/kakacat/minitool/todayinhistory/Presenter.kt
package com.kakacat.minitool.todayinhistory

import com.kakacat.minitool.todayinhistory.model.Article
import com.kakacat.minitool.todayinhistory.model.Model
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

class Presenter(private val view: Contract.View) : Contract.Presenter {
    private val model by lazy { Model() }

    override fun initData() {
        model.initData()
    }

    override fun refreshData() {
        GlobalScope.launch(Dispatchers.Default) {
            val response = model.sendRequest()
            val handleResult = model.handleHistoryResponse(response)
            var result = "请求错误"
            if(handleResult){
                result = "处理成功"
                view.onUpdateDataCallBack(result, true)
            }else{
                view.onUpdateDataCallBack(result)
            }
        }
    }

    override val articleList: List<Article>
        get() = model.getArticleList()

    override val year: Int
        get() = model.year

    override var month: Int
        get() = model.month
        set(month) {
            model.month = month
        }

    override var day: Int
        get() = model.day
        set(day) {
            model.day = day
        }
}